packages/video-http-ingestion/src/VideoHttpIngestionAppliance.js

```diff
@@ -0,0 +1,51 @@
+import { PassThrough } from 'stream'
+import fetch from 'node-fetch'
+import { AbstractVideoIngestionAppliance } from '@tvkitchen/appliance-core'
+
+/**
+ * The VideoHttpIngestionAppliance handles processing a HTTP video stream. It is a concrete
+ * implementation of AbstractIngestionAppliance, which reads in a stream, converts it into
+ * an MPEG-TS stream represented as a series of Payloads, and then emits those Payloads.
+ *
+ * @extends AbstractVideoIngestionAppliance
+ */
+class VideoHttpIngestionAppliance extends AbstractVideoIngestionAppliance {
+	// The URL of the stream to read by the engine
+	url = null
+
+	/**
+	 * Create a VideoHttpIngestionEngine.
+	 *
+	 * @param  {string} settings.url The url of the stream to be ingested
+	 */
+	constructor(settings = {
+		url: '',
+	}) {
+		super(settings)
+		if (!settings.url) {
+			throw new Error(
+				'HttpIngestionAppliance must be instantiated with a stream URL',
+			)
+		}
+		this.url = settings.url
```

Comment:
I'm realizing that the AbstractAppliance already stores passed settings as `this.settings`

Given that, maybe we should just access the setting through that in implementation classes?

(so instead of referencing `this.url` later int he class, we would instead reference `this.settings.url`)

The benefit of this is:

1. It's already doing it.
2. It makes it more explicit that `url` is a setting (as opposed to something generated by the class).

The negative is that it relies on `this.settings` being populated by the super class, which might be a code smell.

Reply:
(As implemented, you would be able to access the url EITHER by:

```
const appliance = new VideoHttpIngestionAppliance({ url: 'yolo' })
console.log(appliance.settings.url)
console.log(appliance.url)
```

Reply:
I'm gonna just rely on the `this.settings`

This also solves the problem of `url` being documented in multiple places.